spotty/providers/aws/deployment/project_resources/bucket/abstruct_bucket.py

```diff
@@ -0,0 +1,25 @@
+from abc import ABC, abstractmethod
+import boto3
+
+from spotty.commands.writers.abstract_output_writrer import AbstractOutputWriter
+
+class AbstractBucketResource(ABC):
+
+    def __init__(self, region: str):
+        self._region = region
+        self._s3 = boto3.client('s3', region_name=region)
+
+    @abstractmethod
+    def get_or_create_bucket(self, output: AbstractOutputWriter, tags: list, dry_run=False):
```

Comment:
Could you please change the type annotation for `tags` to at least `List[Any]`? Also we can add one for `dry_run`.